sycl/include/sycl/ext/oneapi/matrix/matrix-intel.hpp

```diff
@@ -308,6 +322,35 @@ class wi_element<sycl::ext::oneapi::bfloat16, NumRows, NumCols, Use, Layout,
 
 // End wi_element definition
 
+template <typename T, size_t NumRows, size_t NumCols,
+          sycl::ext::oneapi::experimental::matrix::use Use,
+          sycl::ext::oneapi::experimental::matrix::layout Layout,
+          typename Group>
+inline __SYCL_ALWAYS_INLINE std::tuple<uint32_t, uint32_t>
+get_coord(wi_element<T, NumRows, NumCols, Use, Layout, Group> &we) {
+#if defined(__SYCL_DEVICE_ONLY__)
+  __ocl_vec_t<uint32_t, 2> coord =
+      __spirv_JointMatrixGetElementCoordINTEL(we.M.spvm, we.idx);
+  const uint32_t row = coord[0];
+  const uint32_t col = coord[1];
+  return std::make_tuple(row, col);
+#else
```

Comment:
I thought this else is for host side.
You should not need line 348.

Reply:
Good catch.  I just removed it

Reply:
In this case the error txt on line 339 should say same thing as other host side errors (do not mention XMX or AMX).